src/main/java/picocli/CommandLine.java

```diff
@@ -8010,7 +8145,8 @@ public Help(CommandSpec commandSpec, ColorScheme colorScheme) {
             this.aliases.add(0, commandSpec.name());
             this.colorScheme = Assert.notNull(colorScheme, "colorScheme").applySystemProperties();
             parameterLabelRenderer = createDefaultParamLabelRenderer(); // uses help separator
-
+            this.helpFactory = commandSpec.commandLine() != null ? commandSpec.commandLine().getHelpFactory() : new DefaultHelpFactory();
```

Comment:
Not sure if this is the right way to get the helpfactory.

Reply:
Yes, that looks fine. You raise a good point though: the `CommandLine` class is mostly a facade, and it may be good to move some of this down to the `CommandSpec` class. I will go ahead with the merge first, and I may make some changes afterwards.